src/main/java/brotherman/parser/Parser.java

```diff
@@ -0,0 +1,212 @@
+            throw new BrothermanException("Error! There has to be a end date/time!");
+        }
+
+        if (command.split(" ")[0].equals("event") && command.split(" ")[1].equals("/from")) {
+            throw new BrothermanException("Error! There has to be a description");
+        }
+        return true;
+    }
+
+
+    /**
+     * Parses the user input as a todo
+     * @param description Description of the todo
+     * @return Todo parsed from the user input
+     */
+    public static Todo parseTodo(String description) {
+        return new Todo(description);
+    }
+
+
+    /**
+     * Parses the user input as a deadline
+     * @param description Description of the deadline
+     * @param time Due date/time of the deadline
+     * @return Deadline parsed from the user input
+     */
+    public static Deadline parseDeadline(String description, String time) {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(
+                "[yyyy-MM-dd HH:mm:ss][yyyy-MM-dd][MM/dd/yyyy HH:mm:ss][dd/MM/yyyy][d MMM yyyy HHmm]"
+        );
+
+        LocalDateTime dateTimeObj = null;
+
+
+        try {
+            dateTimeObj = LocalDateTime.parse(time, formatter);
+        } catch (DateTimeParseException e) {
+            // If datetime parsing fails, try parsing as date-only
+            try {
+                LocalDate date = LocalDate.parse(time, formatter);
+                dateTimeObj = date.atStartOfDay();
+            } catch (DateTimeParseException ex) {
+                System.out.println("Error parsing datetime: " + ex.getMessage());
+            }
+        }
+
+        return new Deadline(description, dateTimeObj);
+    }
+
+
+    /**
+     * Parses the user input as an event
+     * @param description Description of the event
+     * @param startTime Start date/time of the event
+     * @param endTime End date/time of the event
+     * @return Event parsed from the user input
+     */
+    public static Event parseEvent(String description, String startTime, String endTime) {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(
+                "[yyyy-MM-dd HH:mm:ss][yyyy-MM-dd][MM/dd/yyyy HH:mm:ss][dd/MM/yyyy][d MMM yyyy HHmm]"
+        );
+
+        LocalDateTime startDateTimeObj = null;
+        LocalDateTime endDateTimeObj = null;
+
+
+        try {
+            startDateTimeObj = LocalDateTime.parse(startTime, formatter);
+            endDateTimeObj = LocalDateTime.parse(endTime, formatter);
+        } catch (DateTimeParseException e) {
+            // If datetime parsing fails, try parsing as date-only
+            try {
+                LocalDate startDate = LocalDate.parse(startTime, formatter);
+                LocalDate endDate = LocalDate.parse(endTime, formatter);
+                startDateTimeObj = startDate.atStartOfDay();
+                endDateTimeObj = endDate.atStartOfDay();
+            } catch (DateTimeParseException ex) {
```

Comment:
Im not sure if it is appropriate to handle the exception here. If there is an exception, the programme will continue on with a `null` value returned from this programme. This can cause `NullPointerException` in your programme even when the parse error is already handled here.